src/secret_sharing/decomposed.rs

```diff
@@ -79,6 +83,46 @@ impl<S> BitDecomposed<S> {
             acc + (b * F::truncate_from(1_u128 << i))
         })
     }
+
+    /// Subtraction of two bit-wise secret shares, `self - rhs`, in two's complement.
+    /// Subtracting a value larger than `self` value will cause the result to overflow.
+    /// Be especially careful as the overflow is not checked and could lead to a privacy
+    /// violation (e.g., invalid capping).
+    ///
+    /// # Errors
+    /// If one of the multiplications errors
+    /// # Panics
+    /// If something try to add a bit decomposed value larger than this `SaturatingSum` can accommodate
+    pub async fn sub<C>(
+        &self,
+        ctx: C,
+        record_id: RecordId,
+        rhs: &BitDecomposed<S>,
+    ) -> Result<BitDecomposed<S>, Error>
+    where
+        C: Context,
+        S: LinearSecretSharing<Gf2> + BasicProtocols<C, Gf2>,
+        for<'a> &'a S: LinearRefOps<'a, S, Gf2>,
+    {
+        assert!(self.len() >= rhs.len());
+
+        let mut output = vec![];
+        let mut carry_in = S::share_known_value(&ctx, Gf2::ONE);
+        let zero = S::ZERO;
+        for i in 0..self.len() {
+            let c = ctx.narrow(&BitOpStep::from(i));
+            let compute_carry_out = i < self.len() - 1;
+            let lfs = &self[i];
```

Comment:
lhs

Reply:
"left fand side"

Reply:
fixed :)

Reply:
is this like having a variable called "variable"?